File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    release = []
    for i in range(len(progresses)):
        num = math.ceil((100 - progresses[i])/speeds[i])
        release.append(num)

    cnt = 0
    num = release[0]
    for i in range(len(progresses)):
        if release[i] <= num:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            num = release[i]
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    const days = [];

    for (let i = 0; i < progresses.length; i++) {
        let progress = 100 - progresses[i];
        let speed = speeds[i];
        let tmp = Math.ceil(progress / speed);
        days.push(tmp);
    }
    
    let head = days[0];
    let cnt = 0;
    
    for (let i of days) {
        if (i <= head) {
            cnt++;
        } else {
            answer.push(cnt);
            cnt = 1;
            head = i;
        }
    }
    
    answer.push(cnt);
    return answer;
}
